packages/discovery-react-components/src/components/SearchResults/components/Result/Result.tsx

```diff
@@ -178,18 +178,24 @@ export const Result: React.FunctionComponent<ResultProps> = ({
           />
         ) : (
           <>
-            {displayedText && !showTablesOnlyResults && (
-              <ResultElement
-                body={displayedText}
-                buttonText={hasPassage ? messages.viewExcerptInDocumentButtonText : undefined}
-                element={displayedTextElement}
-                elementType={displayedTextElementType}
-                handleSelectResult={handleSelectResult}
-                passageTextClassName={passageTextClassName}
-                hasResult={!!result}
-                dangerouslyRenderHtml={shouldDangerouslyRenderHtml}
-              />
-            )}
+            {displayedTexts &&
+              !showTablesOnlyResults &&
+              displayedTexts.map((displayedText, index) => {
+                const displayedTextElement = displayedTextElements?.[index];
+                return (
+                  <ResultElement
+                    key={md5(displayedText)}
```

Comment:
It's conceivable that there could be 2+ passages with the same text, which would break this. Using offsets would help (since those are unique) -- but you mentioned to me (elsewhere) that it was difficult to implement.

Reply:
We could probably use the `index` value in conjunction with the hashed text (or hash `text + index`), since the passages can't be re-ordered. This would guarantee uniqueness of the key.